File: CinemaBookingApp/Front-end/CinemaBookingApp/src/types/review.ts
import {ClientInReviewProps} from './client';

export interface ReviewInMovieDetailProps {
  comment: string;
  rating: number;
  createdAt: Date;
  client: ClientInReviewProps;
}

export interface ReviewAddProps {
  movieId: number;
  comment: string;
  rating: number;
}

File: CinemaBookingApp/CinemaBookingApp/CinemaBookingApp/Models/DTOs/Review.cs
using static CinemaBookingApp.Models.DTOs.Client;

namespace CinemaBookingApp.Models.DTOs
{
    public class Review
    {
        public class ReviewInMovieDetailDTO
        {
            public string Comment { get; set; }

            public int Rating { get; set; }

            public DateTime CreatedAt { get; set; }

            public ClientInReviewDTO Client { get; set; }
        }

        public class ReviewCreateDTO
        {
            public long MovieId { get; set; }

            public string Comment { get; set; }

            public int Rating { get; set; }

        }
    }
}
